sklearn/cluster/affinity_propagation_.py

```diff
@@ -90,6 +106,23 @@ def affinity_propagation(S, preference=None, convergence_iter=15, max_iter=200,
     if damping < 0.5 or damping >= 1:
         raise ValueError('damping must be >= 0.5 and < 1')
 
+    preference_array = np.array(preference)
```

Comment:
any reason not to reuse the name `preference`? It's quite clear from how it's used (e.g. `.flat`) that it must be an array.